File: server-go-nebula/shares/response/res.go
package res

type Response struct {
	Ok    bool        `json:"ok"`
	Msg   string      `json:"msg"`
	Data  interface{} `json:"data"`
	Error interface{} `json:"error"`
}

func Ok(msg string, data interface{}, args ...int) Response {
	return Response{true, msg, data, nil}
}

func Err(msg string, args ...int) Response {
	return Response{false, msg, nil, nil}
}

func ErrWithError(msg string, _error interface{}, args ...int) Response {
	return Response{false, msg, nil, _error}
}

func ErrWithData(msg string, data interface{}, args ...int) Response {
	return Response{false, msg, data, nil}
}

File: server-rust-dgraph/src/lib/res.rs
use actix_web::{ HttpResponse };
use crate::lib::errors::{ ServiceError };
use serde_json::{ json, Value as JsonValue };

// 响应体
pub type ServiceResult = anyhow::Result<HttpResponse, ServiceError>;

pub struct Res{}

impl Res{
  // 成功
  pub fn ok(value: JsonValue) -> Result<HttpResponse, ServiceError> {
    let ok_res = json!({
      "ok": true, 
      "status": value.get("status").unwrap_or(&json!(200)),
      "data": value.get("data"), 
      "message": value.get("message")
    });
    Ok(HttpResponse::Ok().json(ok_res))
  }

  // 失败
  pub fn err(value: JsonValue) -> Result<HttpResponse, ServiceError> {
    let err_res = json!({
      "ok": false, 
      "status": value.get("status").unwrap_or(&json!(200)),
      "message": value.get("message"), 
      "errors": value.get("errors")
    });
    Ok(HttpResponse::BadRequest().json(err_res))
  }
}
